### DISCUSSION TITLE
Sub collections

### DISCUSSION THREAD

**Discussion by arcanechimp** [bryanmylee/svelte-headless-table#62] [Q&A]
Is there any way to use hierarchical data using this component? 

What I mean is, every item in the table has a collection of sub items of a different type. I want to preserve the hierarchy in the table, so that the parent items have a row-span set to the size of the child collection, keeping the children grouped with the parent.

Example of what I am trying to achieve:

![image](https://user-images.githubusercontent.com/8119992/193550628-cec6e260-be09-4620-8b1f-584fa396e2dd.png)

Thanks for all your work on this component - it is fantastic!

**Comment by bryanmylee** [comment]
This is something I hadn't considered before, but I can definitely look into it.

My biggest concern right now is making sure it works well with the `addExpandedRows` plugin.

**Comment by bryanmylee** [comment]
Hey @arcanechimp, did you manage to resolve this use case since our previous chat?

**Reply by arcanechimp** [reply]
Hi @bryanmylee, no, no luck yet. 

I'm unsure where to begin. I tried playing around in the REPL with a small set of data back when I first posted, but could not figure out a way using the existing plugins to make it work. 

I'll have another go and get back to you.

**Reply by bryanmylee** [reply]
I'm currently quite busy with work but I'll try my best to help out!

It might take a lot of work to support this feature because right now Svelte Headless Table was built with the assumption that all row models are mapped 1-to-1 to table rows.

**Reply by arcanechimp** [reply]
Yes, that was more or less what I discovered.

I guess it would have to be modeled using parent columns and child columns, along side the parents collection and child collections within them. I can see how ingrained the generic Item class is within the table and plugins, so introducing a new type may cause lots of headaches with the existing plugins.
